src/main/webapp/app/course/plagiarism-cases/instructor-view/plagiarism-cases-instructor-view.component.html

```diff
@@ -49,20 +49,34 @@ <h5 class="mb-0">
                     </div>
                 </div>
                 <div class="card-body">
-                    @for (plagiarismCase of groupedPlagiarismCases[exercise!.id!]; track plagiarismCase) {
+                    @for (plagiarismCase of groupedPlagiarismCases[exercise!.id!]; track plagiarismCase; let j = $index) {
                         <div class="row mb-3">
                             <div class="col-1 text-center">
                                 <a [routerLink]="['/course-management', courseId, 'plagiarism-cases', plagiarismCase.id]">
-                                    {{ plagiarismCase.student.name }} ({{ plagiarismCase.student.login }})
+                                    {{ plagiarismCase.student?.name }} ({{ plagiarismCase.student?.login }})
                                 </a>
                             </div>
                             @if (plagiarismCase.plagiarismSubmissions) {
-                                <div
-                                    class="col-2 text-center"
-                                    jhiTranslate="artemisApp.plagiarism.plagiarismCases.appearsInComparisons"
-                                    [translateValues]="{ count: plagiarismCase.plagiarismSubmissions.length }"
-                                ></div>
+                                <div class="col-2 text-center">
+                                    <span
+                                        id="navigate-plagiarism-detection-{{ i }}-{{ j }}"
+                                        [routerLink]="[
+                                            '/course-management',
+                                            courseId,
+                                            getExerciseUrlSegment(plagiarismCase.exercise!.type),
```

Comment:
Why can we assume that `plagiarismCase.exercise` is not null (as we use the non-null assertion operator here)?

Reply:
Because the groupedPlagiarismCases object only contains exercises that exist. In line 147  of plagiarism-cases-instructor-view.component.ts we do not add an exercise to the groupedPlagiarismCases if exercise or exercise.id is undefined.groupedPlagiarismCasesZ